### PR TITLE
Show categories as annotations on metrics

### PR DESCRIPTION
**Pull request by Lukewh**
## QA

- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/<snap_name>/metrics or https://snapcraft-io-canonical-websites-pr-1493.run.demo.haus/
  (example https://snapcraft-io-canonical-websites-pr-1493.run.demo.haus/slack/metrics)
- Change the period of time a few times

### CODE REVIEW

**Comment by therealjuan** [comment]
This is amazing! Happy new year!

How about adding a ⭐️instead of featured with a tooltip stating `Your snap was featured on DD MMM 201Y`

**Comment by therealjuan** [comment]
What about...

a) ... adding a tooltip stating `Your snap was featured on DD MMM 201Y` ?
b) ... add the category friendly name instead of the category name from the Store and the date in a tooltip like a) ?
c) ... return to a) and b), rinse and repeat?

**Comment by Lukewh** [comment]
@therealjuan adding to the current tooltip, or adding another tooltip?

**Comment by therealjuan** [comment]
Probably better a new tooltip on the `[star] Featured` label

**Comment by bartaz** [comment]
I'm not sure about usefulness of showing 'past' labels like this:

screenshot 2019-02-19 at 15 23 43

At some point in time graphs will all look like this with categories long in the past.
It would maybe be useful to show a label if it was in next period (so it would show up if I choose next period from the dropdown). But showing 'productivity' on a 7d graph when it was set 2 years ago doesn't feel useful or informative.

**Comment by therealjuan** [comment]
👍 on @bartaz

**Comment by therealjuan** [comment]
me > -1

File: webapp/publisher/snaps/views.py [hunk 1]
```diff
@@ -202,23 +204,32 @@ def publisher_snap_metrics(snap_name):
...
+                "name": category["name"],
+            }
 
-        single_series["values"][index_of_date] = 1
+            single_series["values"][index_of_date] = 1
 
-        annotations["series"].append(single_series)
+            annotations["series"].append(single_series)
+        else:
+            new_date = datetime.datetime.strptime(date, "%Y-%m-%d")
```
Comment (bartaz) [inline webapp/publisher/snaps/views.py]: I think we have some logic function to format date that is used somewhere (snap details?). Maybe we could reuse it.

Comment (Lukewh) [reply webapp/publisher/snaps/views.py]: Good catch, thanks @bartaz

**Comment by therealjuan** [comment]
Do we know when a snap is not featured or not part of a category?

* 🗂️`Added to {Category name} in June 2018 / Removed in September 2018`
* ⭐`Featured snap since August 2018 `
* ⭐`Featured snap between August 2018 and September 2018 `

File: static/sass/_snapcraft_metrics.scss [hunk 2]
```diff
@@ -43,4 +46,17 @@
       pointer-events: none;
     }
   }
+
+  .snapcraft-metrics__active-devices {
```
Comment (bartaz) [inline static/sass/_snapcraft_metrics.scss]: Is this used anywhere?

File: webapp/publisher/snaps/views.py [hunk 3]
```diff
@@ -202,6 +202,33 @@ def publisher_snap_metrics(snap_name):
...
+
+        if date not in annotations["buckets"]:
+            annotations["buckets"].append(date)
+
+        index_of_date = annotations["buckets"].index(date)
+
+        single_series = {
+            "values": [0] * (len(annotations)),
+            "name": category["name"],
+            "display_name": category["name"].capitalize().replace("-", " "),
```
Comment (bartaz) [inline webapp/publisher/snaps/views.py]: Not for this PR, but we need a common method to do that. It's already scattered across couple of places both in store and publisher parts.

File: static/sass/_snapcraft_metrics.scss (hunk 2 above)
Comment (Lukewh) [reply static/sass/_snapcraft_metrics.scss]: Ah, now I've made the names go to the left of the | it's not needed, removing thanks

**Pull request merged**